src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -74,11 +74,16 @@ internal static partial class Helpers
                 }
             }
 
-            instructionSetSupportBuilder.ComputeInstructionSetFlags(out var supportedInstructionSet, out var unsupportedInstructionSet,
+            instructionSetSupportBuilder.ComputeInstructionSetFlags(maxVectorTBitWidth, out var supportedInstructionSet, out var unsupportedInstructionSet,
                 (string specifiedInstructionSet, string impliedInstructionSet) =>
                     throw new CommandLineException(string.Format(invalidImplicationMessage, specifiedInstructionSet, impliedInstructionSet)));
 
-            InstructionSetSupportBuilder optimisticInstructionSetSupportBuilder = new InstructionSetSupportBuilder(targetArchitecture);
+            // Due to expansion by implication, the optimistic set is most often a pure superset of the supported set
+            //
+            // However, there are some gaps in cases like Arm64 neon where none of the optimistic sets imply it. Likewise,
+            // the optimistic set would be missing the explicitly unsupported sets. So we effectively clone the list and
+            // tack on the additional optimistic bits after. This ensures the optimistic set remains an accurate superset
+            InstructionSetSupportBuilder optimisticInstructionSetSupportBuilder = new InstructionSetSupportBuilder(instructionSetSupportBuilder);
```

Comment:
@davidwrighton, this seems like it was an existing bug and/or quirk.

The scenario seems to be that the user may have explicitly opted out of an instruction set, such as `-popcnt`. However, since `getJitFlags` just does `flags.InstructionSetFlags.Add(_compilation.InstructionSetSupport.OptimisticFlags);`, that meant that the falgs the JIT saw would include `popcnt` anyways. The actual `ExplicitlyUnsupportedFlags` is only used as part of `ToInstructionSetSupportString`, which does impact loadability of various JIT bits. However, the JIT would have still been marking the methods as using `popcnt` due to the `compOpportunisticallyDependsOn` reporting `true` causing us to throw away based on non meaningful JIT/corelib optimizations.

There was also the quirk that the opportunistic set didn't include `neon` for Arm64, which led to an assert where I was trying to validate it was always present due to being "baseline".